src/libs/ajax/leonardo/Disks.ts

```diff
@@ -5,29 +5,46 @@ import { DecoratedPersistentDisk, GetDiskItem, ListDiskItem } from 'src/libs/aja
 import { updatePdType } from 'src/pages/workspaces/workspace/analysis/utils/disk-utils'
 
 
-export const Disks = signal => ({
-  list: async (labels = {}): Promise<ListDiskItem> => {
-    const res = await fetchLeo(`api/google/v1/disks${qs.stringify(labels, { addQueryPrefix: true })}`,
-      _.mergeAll([authOpts(), appIdentifier, { signal }]))
-    return res.json()
-  },
+export const Disks = signal => {
+  const v2Func = () => {
+    const diskV2Root = 'api/v2/disks/'
+    return {
+      delete: (workspaceId, diskId): Promise<void> => {
+        return fetchLeo(`${diskV2Root}/${workspaceId}/${diskId}`, _.mergeAll([authOpts(), appIdentifier, { signal, method: 'DELETE' }]))
+      }
+    }
+  }
+
+  const v1Func = () => {
```

Comment:
Talking out loud:

Hmm... a little torn between having v1Func take no args vs having it take (name, project), and having list at a higher level. I'm fine with the current setup, as it simplifies having only 2 top-level entities (v1/v2), which makes sense semantically. However, it hides the fact that v1 largely deals with (name,project) and v2 deals with workspaceId down a layer for the sake of nesting list under v1/v2

Reply:
I want to change the previous implementation as little as possible. Realistically, list should optionally take the google project in v1.

Personally, I don't see much of a need to wrap CRUD with disk. I'd like to see v1Func flat with the actions and each take the args they need.

Reply:
As for v2, I personally think the individual functions should use the arguments they need. I don't think it's necessary to create the v2 func with any args, just use the function with what they need.